Fix #74: 根据issues中的问题，我对代码的一些改进

mlibio opened the pull request:
实现了如下功能：
- 将点赞数据在 redis 中存储。
- 创建了一个定时任务类：LikeDataSyncScheduler。
- 使用Spring的@Scheduled注解创建了每小时执行一次的定时任务。
- 定时任务从Redis中读取帖子的点赞增量，并更新到数据库中。
- 更新完成后，清除已处理过的Redis缓存数据。

mlibio committed:
feat: 添加每小时刷新帖子点赞量的定时任务

实现了如下功能：
- 将点赞数据在 redis 中存储。
- 使用Spring的@Scheduled注解创建了每小时执行一次的定时任务。
- 定时任务从Redis中读取帖子的点赞增量，并更新到数据库中。
- 更新完成后，清除已处理过的Redis缓存数据。

mlibio committed:
feat: 添加每小时刷新帖子点赞量的定时任务

实现了如下功能：
- 将点赞数据在 redis 中存储。
- 使用Spring的@Scheduled注解创建了每小时执行一次的定时任务。
- 定时任务从Redis中读取帖子的点赞增量，并更新到数据库中。
- 更新完成后，清除已处理过的Redis缓存数据。

Durancer commented on post-service/post-server/src/main/java/com/xueyu/post/service/impl/PostOperateServiceImpl.java:
```diff
@@ -62,7 +68,9 @@ public Boolean likeUserPost(Integer postId, Integer userId) {
-			postGeneralMapper.updateLikeNumByPostId(postId, 1);
+			//postGeneralMapper.updateLikeNumByPostId(postId, 1);
+			//先将数据在 redis 中存储，使用定时任务刷新点赞数，并更新到数据库中
+			redisTemplate.opsForHash().increment(CACHE_POST_LIKES_KEY, postId, 1);
```
这里考虑了增加的情况，如果帖子被取消点赞，应该要有缓存或db扣减的逻辑，这里缺少取消点赞时的处理

Durancer replied:
这里我建议：
1、减少时，先判断缓存有无点赞数量积累，如有，则扣缓存，否则扣db中的值
2、可以将redis用弱关联使用，即在redis 因各种原因不可用时，可以直接更新db，保持最终一致性即可，如 try catch起来
3、未来如有相关数据查询接口，有个todo 时，查表时需要在查一边redis，将数据累加，得到正确的结果

Durancer commented on post-service/post-server/src/main/java/com/xueyu/post/task/LikeDataSyncScheduler.java:
```diff
@@ -0,0 +1,45 @@
+        this.hashOps = hashOps;
+    }
+
+    // 每小时执行一次
```
方法注释最好使用 doc文档注释，可以参考其他方法

Durancer commented on post-service/post-server/src/main/java/com/xueyu/post/task/LikeDataSyncScheduler.java:
```diff
@@ -0,0 +1,45 @@
+
+@Component
+@Slf4j
+public class LikeDataSyncScheduler {
```
在类上最好也补充doc注释，可以参考其他类

Durancer reviewed (commented):
hii, mlibio! 
非常高兴收到你的贡献，我大概看了你的代码，写了一些建议的评论
你可以看看对应的地方做一些修改，
然后可以重新发起一个 PR 请求，（一般不直接向main发起请求）
我在仓库为你创建了refresh-post-likes-hourly 分支，你可以向这个分支发起合并请求
建议在提交pr后，在评论区贴两张自测的图片。

mlibio committed:
增加定时任务处理点赞数据及优化取消点赞逻辑

1. 增加了定时任务类和相关的文档注释，用于定期同步Redis中的点赞数据到数据库。
2. 在原有基础上增加了取消点赞时的操作逻辑，确保在Redis中存在点赞记录时优先递减Redis中的计数，否则更新数据库。
3. 使用try-catch块来弱关联Redis操作，当Redis不可用时直接更新数据库，保持最终一致性。
4. 建议在实现统计帖子点赞数等与点赞数据强相关的接口时，先同步Redis中的点赞数据变更到数据库，以确保查询结果的准确性。

mlibio commented:
你好，感谢你的建议
我已根据你的建议修改了代码，已经提交了新的PR
但由于我没有数据库，无法进行测试，希望它能帮到你
